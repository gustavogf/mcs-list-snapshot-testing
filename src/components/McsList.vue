<template>
  <div>
    <ul class="list">
      <li class="list__item" v-for="mc in filteredList">{{ mc }}</li>
    </ul>
  </div>
</template>

<script>
export default {
  props: {
    query: {
      type: String,
      default: 'MC',
    },
  },
  computed: {
    filteredList() {
      return this.list.filter(mcName => {
        return mcName.toLowerCase().includes(this.query.toLowerCase())
      });
    },
  },
  data() {
    return {
      list: [
        'Don Juan',
        'Fioti',
        'Jerry Smith',
        'Kekel',
        'Kevinho',
        'Lan',
        'Loma',
        'MM',
        'Nego do Borel',
        'WM',
      ],
    };
  },
};
</script>

<style scoped>
.list {
  list-style: none;
  margin: 0 auto;
  max-width: 300px;
  padding-left: 0;
}

.list__item {
  background-color: #eee;
  border-radius: 8px;
  border: 2px solid darkgray;
  font-size: 18px;
  font-weight: 600;
  margin-bottom: 16px;
  padding: 10px 15px;
  text-align: center;
}
</style>
