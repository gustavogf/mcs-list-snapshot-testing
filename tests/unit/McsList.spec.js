import { shallowMount } from '@vue/test-utils'
import McsList from '@/components/McsList.vue'

describe('McsList.vue', () => {
  it('Renders filtered list', () => {
    const query = 'ke'
    const mcsList = shallowMount(McsList, {
      propsData: { query }
    })

    expect(mcsList.html()).toMatchSnapshot()
  })
})
