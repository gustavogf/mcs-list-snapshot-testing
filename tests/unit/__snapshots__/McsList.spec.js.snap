// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`McsList.vue Renders filtered list 1`] = `
<div>
  <ul class="list">
    <li class="list__item">Kekel</li>
    <li class="list__item">Kevinho</li>
  </ul>
</div>
`;
